contrib/Intel/JenkinsBuildScripts/tests.py

```diff
@@ -0,0 +1,311 @@
+
+class fabtest(test):
+    
+    def __init__(self, branchname, buildno, testname, core_prov, fabric, hosts, util_prov=None):
+        super().__init__(branchname, buildno, testname, core_prov, fabric, hosts, util_prov)
+        self.fabtestpath = "{}/bin".format(self.libfab_installpath) 
+   
+    @property    
+    def cmd(self):    
+        #return "./runfabtests.sh "
+        return "{}/runfabtests.sh ".format(self.fabtestpath)
+     
+    @property
+    def options(self):
+        opts = "-vvv -p {} -S ".format(self.fabtestpath)
+        if (self.core_prov == "verbs" and self.nw_interface):
+            opts = "{} -s {} ".format(opts, common.get_node_name(self.server, self.nw_interface)) # include common.py
+            opts = "{} -c {} ".format(opts, common.get_node_name(self.client, self.nw_interface)) # from common.py
+        if (self.core_prov == "shm"):
+            opts = "{} -s {} ".format(opts, self.server)
+            opts = "{} -c {} ".format(opts, self.client)
+            opts += "-N "
+            
+        if not re.match(".*sockets|udp|tcp.*", self.core_prov):
+            opts = "{} -t all ".format(opts)
+
+        efile = self.get_exclude_file()
+        if efile:
+            opts = "{} -R ".format(opts)
+            opts = "{} -f {} ".format(opts, efile)  
+        
+        for key,val in self.env:
+            opts = "{options} -E {key}={value} ".format(options = opts, key=key, value=val)
+    
+        if self.util_prov:
+            opts = "{options} {core};{util} ".format(options=opts, core=self.core_prov, util=self.util_prov)
+        else:
+            opts = "{options} {core} ".format(options=opts,core=self.core_prov)
+        
+        if (self.core_prov == "shm"):
+            opts += "{} {} ".format(self.client, self.server)
+        else:
+            opts += "{} {} ".format(self.server, self.client)
+             
+        return opts
+   
+    @property
+    def execute_condn(self):
+        return True
+
+    def execute_cmd(self):
+ 
+        output = self.cmd + self.options
+        outputcmd = shlex.split(output)
+        print(outputcmd)
+        p = subprocess.Popen(outputcmd,stdout=subprocess.PIPE, text=True)
+        while True:
+            out = p.stdout.read(1)
+            if (out == "" and p.poll() != None):
+                break
+            if (out != ""):
+                sys.stdout.write(out)
+                sys.stdout.flush()
+        
+        if (p.returncode != 0):
+            print("exiting with " + str(p.poll()))
+            sys.exit(p.returncode)
+
+        
+
+class mpitests(test):
+    def __init__(self, branchname, buildno, testname, core_prov, fabric, mpitype, hosts, util_prov=None):
+        super().__init__(branchname, buildno, testname, core_prov, fabric, hosts, util_prov)
+        self.mpi = mpitype
+
+
+    @property
+    def cmd(self):
+        if (self.mpi == "impi" or self.mpi == "mpich"):
+            self.testpath = "/home/build/ssg_sfi-buildbot/scripts"
```

Comment:
Absolute path prefixes should come from site_config. Ditto for other path references elsewhere.

Reply:
what about path references which are partially constructed dynamically and partially use a constant prefix for e.g.:
{}/share/fabtests/test_configs".format(path) where path is constructed dynamically since it requires the branch name and build number

Reply:
Dynamically constructed ones are fine. Only the absolute references like "/home/build" should be hidden in site_config.